packages/server-ct/src/socket-ct.ts

```diff
@@ -33,26 +34,30 @@ const reporterEvents = [
   'reporter:restarted',
 ]
 
-const retry = (fn) => {
+const retry = (fn: (res: any) => void) => {
   return Bluebird.delay(25).then(fn)
 }
 
 class Socket {
-  constructor (config) {
+  private ended: boolean
+  private io: SocketIO.Server
+  private testsDir: string[]
+
+  constructor (config: Record<string, any>) {
     this.ended = false
 
     this.onTestFileChange = this.onTestFileChange.bind(this)
   }
 
-  onTestFileChange (filePath) {
+  onTestFileChange (filePath: string) {
     debug('test file changed %o', filePath)
   }
 
-  toReporter (event, data) {
+  toReporter (event: string, data: any) {
```

Comment:
Would it make sense to define a set of enums for the valid events?

Reply:
sure would, is there an easy way to get a list of all valid events for this one?

Reply:
I think here:

https://github.com/cypress-io/cypress/pull/9625/files#diff-0ea726fd88825aefe4de2a52bbdddccfe1bca5436607f3b29fd93bf5840e05abR14-L35